ARTstor.js

```diff
@@ -34,98 +37,1043 @@
+    }
+
+    return newS;
+}
+
+
+encodeMap = {
+    ' ': '20',
+    '!': '21',
+    '"': '22',
+    '#': '23',
+    '$': '24',
+    '%': '25',
+    '&': '26',
+    "'": '27',
+    '(': '28',
+    ')': '29',
+    '*': '2A',
+    '+': '2B',
+    ',': '2C',
+    '-': '2D',
+    '.': '2E',
+    '/': '2F',
+    '0': '30',
+    '1': '31',
+    '2': '32',
+    '3': '33',
+    '4': '34',
+    '5': '35',
+    '6': '36',
+    '7': '37',
+    '8': '38',
+    '9': '39',
+    ':': '3A',
+    ';': '3B',
+    '<': '3C',
+    '=': '3D',
+    '>': '3E',
+    '?': '3F',
+    '[': '5B',
+    //				'\\' : '5C',
+    ']': '5D',
+    '^': '5E',
+    '_': '5F',
+    '`': '60',
+    '{': '7B',
+    '|': '7C',
+    '}': '7D',
+    '~': '7E'
+};
+
+
+function processIVURL(text1, artItem, serverURL, url) {
+    Zotero.debug("processIVURL  text " + text1);
+    var text = text1;
+    var service = text.substring(text.indexOf("secure"));
+
+    service = service.substring(0, service.indexOf("</td>"));
+    //	Zotero.debug("processIVURL service1 "+service);
+    service = service.replace(/<wbr\/>/g, "");
+    service = service.substring(service.indexOf("?"));
+    service = service.trim();
+    var ivURL = serverURL + "secure/ViewImages" + service + "&zoomparams=&fs=true";
+
+    if (url == "") {
+        artItem.url = ivURL;
+    } else {
+        //	artItem.url=url;
+        artItem.url = ivURL;
+    }
+
+    Zotero.debug("processIVURL service2 ivURL " + ivURL);
+}
+
+function getMetadata(itemsArray, serverURL, url, doc) {
+    var serviceURL4Metadata = serverURL + "secure/metadata/";
+
+
+    (function next() {
+        if (!itemsArray.length) return;
+        var objectIdMod = itemsArray.shift();
+        var items = objectIdMod.split(":");
+        var item = items[0];
+        var objectTypeId = items[1];
+        Zotero.debug("getMetadata  service URL  " + serviceURL4Metadata + item);
+        Zotero.debug("getMetadata  objectTypeId  " + objectTypeId);
+        Zotero.Utilities.HTTP.doGet(serviceURL4Metadata + item, function(text) {
+
+
+            var artItem = processMetadata(text, item, doc, serverURL);
+            artItem.attachments.push({
+                title: "Artstor Thumbnails",
+
+                document: doc
+
+            });
+            var server = serverURL.substring(0, serverURL.length - 8);
+            switch (objectTypeId) {
+                case "13":
+                case "10":
+                    var serviceURL4IVURL = serviceURL4Metadata + item + "?_method=FpHtml";
+                    Zotero.Utilities.HTTP.doGet(serviceURL4IVURL, function(text) {
+                        processIVURL(text, artItem, serverURL, url);
+                    });
+
+                    break;
+                case "11":
+                    var serviceURL4QTVR = serverURL + "secure/imagefpx/" + item + "/7"; //QTVR
+                    Zotero.Utilities.HTTP.doGet(serviceURL4QTVR, function(text) {
+                        var jsonQTVR = eval('(' + text + ')');
```

Comment:
That's no good. Can't use `eval` just like that (it's not secure). Seems like you just need `JSON.parse()`